Website/api/index.js

```diff
@@ -0,0 +1,493 @@
+  let jsonData = JSON.parse(rawdata);
+
+  let d = String(req.params.date);
+
+  var dd = new Date(d);
+  var firstDate = new Date(Object.keys(jsonData["dates"])[0]);
+  var lastDate = new Date(Object.keys(jsonData["dates"])[Object.keys(jsonData["dates"]).length - 1]);
+
+  let draft = {"type": "overview", "data": { "dateText": updateDate(d), "blocks": []}};
+
+  if (dd.getDay() === 6 || dd.getDay() === 0 || jsonData["dates"][String(d)] === 11 || dd < firstDate || dd > lastDate) {
+    draft.data.blocks.push(template("nodata"));
+    res.send(JSON.stringify(draft));
+    return;
+  }
+
+  draft.data.blocks.push(template("recessBanner"));
+
+  var recessDuty = jsonData["presets"][jsonData["dates"][String(d)]]["recess_duty"];
+  var morningDutyData = jsonData["constants"][jsonData["presets"][jsonData["dates"][String(d)]]["constant_morning_duty"]];
+  var flagDuty = overviewFlagDuty(d);
+
+  var recessBadge = {
+    1: {"colour": "bg-primary", "text": "First Recess (9:10 - 9:45)"},
+    2: {"colour": "bg-secondary", "text": "Second Recess (9:45 - 10:20)"},
+    3: {"colour": "bg-success", "text": "Third Recess (10:20 - 10:55)"},
+    4: {"colour": "bg-info", "text": "Fourth Recess (10:55 - 11:30)"}
+  };
+
+  var dutyTrans = {
+    nb1: "NB1",
+    nb2: "NB2",
+    nb3: "NB3",
+    stall: "Stall",
+    pond: "Pond",
+    field: "Field",
+    parade_square: "Parade Square",
+    lift: "Lift",
+    central: "Central",
+    patrol: "Patrol",
+    on_break: "On Break",
+    main_gate: "Main Gate",
+    late_coming: "Late Coming",
+    canteen: "Canteen",
+    stairs_to_hall: "Stairs To Hall",
+    side_gate: "Side Gate",
+    morning_patrol: "Morning Patrol",
+    hall: "Hall",
+  };
+
+  let counter1 = 1;
+
+  let ongingRecess = 0;
+
+  if (checkTimeValid("09:10:00", "11:30:00") && today(String(d))) {
+    if (checkTimeValid("09:10:00", "09:45:00")) {
+        ongingRecess = 1;
+    } else if (checkTimeValid("09:45:00", "10:20:00")) {
+        ongingRecess = 2;
+    } else if (checkTimeValid("10:20:00", "10:55:00")) {
+        ongingRecess = 3;
+    } else if (checkTimeValid("10:55:00", "23:30:00")) {
+        ongingRecess = 4;
+    }
+}
+
+  for (const recess in recessDuty){
+    let active = false;
+
+    if (ongingRecess !== 0 && counter1 === ongingRecess) {
+        active = true;
+    }
+
+    let draft1 = {"type": "recessBlock", "active": active, "blocks": []};
+
+    draft1.blocks.push({"type": "recessTitle", "colour": recessBadge[counter1].colour, "text": recessBadge[counter1].text});
+    for (const items in recessDuty[recess]) {
+        draft1.blocks.push({"type": "dutySpots", "title": dutyTrans[items], "text": listToString(recessDuty[recess][items])});
+    }
+    counter1 += 1;
+
+    draft.data.blocks.push(draft1);
+}
+
+let draft2 = {"type": "morningBlock", "blocks": []};
+
+draft2.blocks.push({"type": "flagRaisingDuty", "singaporeFlag": flagDuty['singapore_flag'], "schoolFlag": flagDuty["school_flag"]});
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Davianlols6/SC-Duty-List/pullRequest?prid=8404081)
```suggestion
draft2.blocks.push({"type": "flagRaisingDuty", "singaporeFlag": flagDuty["singapore_flag"], "schoolFlag": flagDuty["school_flag"]});
```